packages/z/zbar/port/config.h

```diff
@@ -0,0 +1,480 @@
+
+/* Define to 1 to use dbus */
+/* #undef HAVE_DBUS */
+
+/* Define if the GNU dcgettext() function is already present or preinstalled.
+   */
+#define HAVE_DCGETTEXT 1
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#define HAVE_DLFCN_H 1
+
+/* Define to 1 if you have the <errno.h> header file. */
+#define HAVE_ERRNO_H 1
+
+/* Define to 1 if you have the <fcntl.h> header file. */
+#define HAVE_FCNTL_H 1
+
+/* Define to 1 if you have the <features.h> header file. */
+#define HAVE_FEATURES_H 1
+
+/* Define to 1 if you have the <float.h> header file. */
+#define HAVE_FLOAT_H 1
+
+/* Define to 1 if you have the 'floor' function. */
+/* #undef HAVE_FLOOR */
+
+/* Define to 1 if fseeko (and ftello) are declared in stdio.h. */
+/* #undef HAVE_FSEEKO */
+
+/* Define to 1 if you have the 'getcwd' function. */
+#define HAVE_GETCWD 1
+
+/* Define to 1 if you have the 'getpagesize' function. */
+#define HAVE_GETPAGESIZE 1
+
+/* Define if the GNU gettext() function is already present or preinstalled. */
+#define HAVE_GETTEXT 1
+
+/* Define to 1 if you have the 'gettimeofday' function. */
+#define HAVE_GETTIMEOFDAY 1
+
+/* Define to 1 to use GraphicsMagick */
+/* #undef HAVE_GRAPHICSMAGICK */
+
+/* Define if you have the iconv() function and it works. */
+#define HAVE_ICONV 1
+
+/* Define to 1 to use ImageMagick */
+/* #undef HAVE_IMAGEMAGICK */
+
+/* Define to 1 to use ImageMagick 7 */
+/* #undef HAVE_IMAGEMAGICK7 */
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#define HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the <jerror.h> header file. */
+/* #undef HAVE_JERROR_H */
+
+/* Define to 1 if you have the <jni.h> header file. */
+/* #undef HAVE_JNI_H */
+
+/* Define to 1 if you have the <jpeglib.h> header file. */
+/* #undef HAVE_JPEGLIB_H */
+
+/* Define to 1 if you have the 'jpeg' library (-ljpeg). */
+/* #undef HAVE_LIBJPEG */
+
+/* Define to 1 if you have the 'pthread' library (-lpthread). */
+#define HAVE_LIBPTHREAD 1
+
+/* Define to 1 if you have the <libv4l2.h> header file. */
+/* #undef HAVE_LIBV4L2_H */
+
+/* Define to 1 if you have the <limits.h> header file. */
+#define HAVE_LIMITS_H 1
+
+/* Define to 1 if you have the <linux/videodev2.h> header file. */
+/* #undef HAVE_LINUX_VIDEODEV2_H */
+
+/* Define to 1 if you have the <linux/videodev.h> header file. */
+/* #undef HAVE_LINUX_VIDEODEV_H */
+
+/* Define to 1 if you have the 'localeconv' function. */
+#define HAVE_LOCALECONV 1
+
+/* Define to 1 if you have the <locale.h> header file. */
+#define HAVE_LOCALE_H 1
+
+/* Define to 1 if you have the <MagickWand/MagickWand.h> header file. */
+/* #undef HAVE_MAGICKWAND_MAGICKWAND_H */
+
+/* Define to 1 if you have the 'malloc' function. */
+#define HAVE_MALLOC 1
+
+/* Define to 1 if you have the <malloc.h> header file. */
+/* #undef HAVE_MALLOC_H */
+
+/* Define to 1 if you have the 'memchr' function. */
+#define HAVE_MEMCHR 1
+
+/* Define to 1 if you have the 'memmove' function. */
+#define HAVE_MEMMOVE 1
+
```

Comment:
we should use add_configfiles + configvar

https://github.com/xmake-io/xmake-repo/blob/5ca5ddc2eb11f8efe35a683a5e4dc701a707d4b3/packages/l/libiconv/port/xmake.lua#L229

Reply:
I will try `io.gsub("config.h.in", "# ?undef (.-)\n", "${define %1}\n")`